From 82e24b8d7c3f90f138f3f76d9e4a921d55f52851 Mon Sep 17 00:00:00 2001
From: Yuxuan Shui <yshui@codeweavers.com>
Date: Fri, 13 Jun 2025 18:51:59 +0100
Subject: [PATCH 021/293] shell32: Fix use-after-free in ShellView_WndProc.

RevokeDragDrop release the drop target, if that happens to be the shell view
itself, doing so will cause pThis to be freed.
---
 dlls/shell32/shlview.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/shell32/shlview.c b/dlls/shell32/shlview.c
index a39d51dae2a..b1524b1f29f 100644
--- a/dlls/shell32/shlview.c
+++ b/dlls/shell32/shlview.c
@@ -1696,8 +1696,8 @@ static LRESULT CALLBACK ShellView_WndProc(HWND hWnd, UINT uMessage, WPARAM wPara
           case WM_GETFONT:      return SendMessageW(pThis->hWndList, WM_GETFONT, wParam, lParam);
 
 	  case WM_DESTROY:	
-	  			RevokeDragDrop(pThis->hWnd);
 				SHChangeNotifyDeregister(pThis->hNotify);
+				RevokeDragDrop(pThis->hWnd);
 	                        break;
 
 	  case WM_ERASEBKGND:
-- 
2.43.0

